rfcs/48-diverse-defaults.md

```diff
@@ -0,0 +1,60 @@
+# Feature Name: Different plugin groups to support different use cases (diverse-defaults for short)
+
+## Summary
+
+There are many different categories of bevy applications. Each use case has a sufficiently different commonsense default configuration to warrant different use case specific default plugins.
+
+## Motivation
+
+Bevy would be doing this to support easier and faster 'getting started' user story for non-minimal bevy applications. It will be particularly helpful to beginners but would also be a nice ergonomics/UX win for experienced bevy devs to not have to configure every minute detail manually when setting up a new project that is in one of the above outlined application domains.
+
+## User-facing explanation
+
+Where there used to just be `DefaultPlugins` and `MiniminalPlugins` there will now be many different `DefaultPlugins` a user can choose from that exist for many common categories of bevy applications.
+
+Bevy now provides you a number of convenient `PluginGroups` to help you get your application started with sensible defaults for your use case. The current list is:
+
+- `MinimalPlugins`: The very basic internal structure required to ensure things like `Time` function properly.
+- `GamePlugins`: Suitable for making both 2D and 3D real-time games with input handling, audio and rendering (note: previously `DefaultPlugins`).
+- `2dGamePlugins`: Like `GamePlugins`, but limited to the systems and resources needed in a 2D game.
+- `TuiPlugins`: Suitable for making TUI applications.
```

Comment:
This needs a note in implementation details on what we should include to get this working.

Reply:
What exactly would we need to include to get this working? I don't understand the details of the bevy `DefaultPlugins` code I guess.

Reply:
Also would probably prefer to just migrate my initial proposal to the one @mockersf suggested which I think addresses the problem area even better! :)

Reply:
The bigger problem is "I don't know how to make a terminal game in Bevy 0.6" at all. There were a few ecosystem crates, but they're no longer up to date.